core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -146,64 +146,65 @@ class BrokerTopicMetrics(name: Option[String]) extends KafkaMetricsGroup {
-    if (name.isEmpty) Some(metricTypeMap.getAndMaybePut(
-      BrokerTopicStats.ReplicationBytesInPerSec,
-      newMeter(BrokerTopicStats.ReplicationBytesInPerSec, "bytes", TimeUnit.SECONDS, tags)))
+    if (name.isEmpty) Some(metricTypeMap(BrokerTopicStats.ReplicationBytesInPerSec).meter())
     else None
   private[server] def replicationBytesOutRate =
-    if (name.isEmpty) Some(metricTypeMap.getAndMaybePut(
-      BrokerTopicStats.ReplicationBytesOutPerSec,
-      newMeter(BrokerTopicStats.ReplicationBytesOutPerSec, "bytes", TimeUnit.SECONDS, tags)))
+    if (name.isEmpty) Some(metricTypeMap(BrokerTopicStats.ReplicationBytesOutPerSec).meter())
     else None
-  def failedProduceRequestRate = metricTypeMap.getAndMaybePut(BrokerTopicStats.FailedProduceRequestsPerSec,
-    newMeter(BrokerTopicStats.FailedProduceRequestsPerSec, "requests", TimeUnit.SECONDS, tags))
-  def failedFetchRequestRate = metricTypeMap.getAndMaybePut(BrokerTopicStats.FailedFetchRequestsPerSec,
-    newMeter(BrokerTopicStats.FailedFetchRequestsPerSec, "requests", TimeUnit.SECONDS, tags))
-  def totalProduceRequestRate = metricTypeMap.getAndMaybePut(BrokerTopicStats.TotalProduceRequestsPerSec,
-    newMeter(BrokerTopicStats.TotalProduceRequestsPerSec, "requests", TimeUnit.SECONDS, tags))
-  def totalFetchRequestRate = metricTypeMap.getAndMaybePut(BrokerTopicStats.TotalFetchRequestsPerSec,
-    newMeter(BrokerTopicStats.TotalFetchRequestsPerSec, "requests", TimeUnit.SECONDS, tags))
-  def fetchMessageConversionsRate = metricTypeMap.getAndMaybePut(BrokerTopicStats.FetchMessageConversionsPerSec,
-    newMeter(BrokerTopicStats.FetchMessageConversionsPerSec, "requests", TimeUnit.SECONDS, tags))
-  def produceMessageConversionsRate = metricTypeMap.getAndMaybePut(BrokerTopicStats.ProduceMessageConversionsPerSec,
-    newMeter(BrokerTopicStats.ProduceMessageConversionsPerSec, "requests", TimeUnit.SECONDS, tags))
+  def failedProduceRequestRate = metricTypeMap(BrokerTopicStats.FailedProduceRequestsPerSec).meter()
+  def failedFetchRequestRate = metricTypeMap(BrokerTopicStats.FailedFetchRequestsPerSec).meter()
+  def totalProduceRequestRate = metricTypeMap(BrokerTopicStats.TotalProduceRequestsPerSec).meter()
+  def totalFetchRequestRate = metricTypeMap(BrokerTopicStats.TotalFetchRequestsPerSec).meter()
+  def fetchMessageConversionsRate = metricTypeMap(BrokerTopicStats.FetchMessageConversionsPerSec).meter()
+  def produceMessageConversionsRate = metricTypeMap(BrokerTopicStats.ProduceMessageConversionsPerSec).meter()
 
   // this method helps check with metricTypeMap first before deleting a metric
-  def removeMetricHelper(metricType: String, tags: scala.collection.Map[String, String]): Unit = {
-    val metric: Meter = metricTypeMap.remove(metricType)
-    if (metric != null) {
-      removeMetric(metricType, tags)
-    }
-  }
+  def removeMetricHelper(metricType: String, tags: scala.collection.Map[String, String]): Unit =
+    metricTypeMap.remove(metricType).foreach(_.close())
```

Comment:
If you remove a metric via this method, how are they added back? I think you cannot remove it from `metricTypeMap`, you have to:
1. Call `close()`
2. Change `close()` to set its var to `null` so that if it's invoked again, it's recreated.